src/psyclone/psyir/nodes/node.py

```diff
@@ -1094,6 +1095,48 @@ class or a tuple of classes) is found. This can be used to avoid
             local_list += child.walk(my_type, stop_type)
         return local_list
 
+    def split_consecutive(self, my_type, stop_type=None):
```

Comment:
I'm not sure whether the name `split_consecutive` is the best choice. The operation does feel like Python string splitting but the name doesn't indicate that there will be any `walk`ing involved. I also considered a name including "twins" (immediate siblings). Open to suggestions!